internal/pkg/runtime/engines/singularity/container_linux.go

```diff
@@ -531,6 +531,15 @@ func (c *container) mountGeneric(mnt *mount.Point) (err error) {
 		}
 	}
 	_, err = c.rpcOps.Mount(source, dest, mnt.Type, flags, optsString)
+	// when using user namespace we always try to apply mount flags,
+	// then if we get a permission denied error, we contine execution
+	// by ignoring the error and warn user
+	if remount && err != nil && c.userNS {
```

Comment:
Note that err != nil is redundant here again.

Reply:
this is required otherwise the call `err.Error` will panic when `err` is nil, I reverted change in `addUserbindsMount` too